purchase_security/models/res_users.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    purchase_team_ids = fields.Many2many(
+        comodel_name="purchase.team",
+        relation="purchase_team_res_users_rel",
+        column1="res_users_id",
+        column2="purchase_team_id",
+        string="Purchases Teams",
+        check_company=True,
+        copy=False,
+        readonly=True,
+    )
+    purchase_team_id = fields.Many2one(
```

Comment:
Why do you need this field instead using `purchase_team_ids[:1]`?

Reply:
This field is required to use it in https://github.com/OCA/purchase-workflow/pull/2153/files#diff-dfb94a4f11b575139dd2bfa579f9a012280cdc3ad65e43aa10e222113209ae2bR42 do you think it could be done in another way?

Reply:
My suggestion about using `purchase_team_ids[:1]`. Anyway, I don't quite understand the rule adding all the users of the first team of the user.

Reply:
The "_User (team orders)_" permission added by this module will allow to see the partners that have set the user's purchase team. 

There is no `purchase_team_id` field in the partners to be able to filter by that field, so the rule adds the domain to be able to see the partners linked to users of the same purchase team of the user.

Reply:
I'm still talking about the user, not the partner. Just use `user.purchase_team_ids[:1]`